<?xml version="1.0"?>
<launch>
  <arg name="run_rviz" default="true" />
  <node name="itf" pkg="rviz_interactive_tf" type="interactive_tf" ns="itf1">
    <param name="parent_frame" value="map"/>
    <param name="frame" value="tf1"/>
    <!-- set initial value -->
    <param name="x" value="3.0"/>
    <param name="y" value="1.0"/>
    <param name="z" value="0.0"/>
    <param name="roll" value="0.0"/>
    <param name="pitch" value="0.3"/>
    <param name="yaw" value="0.0"/>
  </node>
  <!-- when these are chained together, the marker controls don't
      appear unless I switch rviz to the tf1 frame, then move tf2,
      then I can go back to the map frame and see tf2 markers -->
  <!-- <node name="itf" pkg="rviz_interactive_tf" type="interactive_tf" ns="itf2"> -->
  <!--   <param name="parent_frame" value="tf1"/> -->
  <!--   <param name="frame" value="tf2"/> -->
  <!-- </node> -->
  <node name="itf" pkg="rviz_interactive_tf" type="interactive_tf" ns="itf2">
    <param name="parent_frame" value="base_footprint"/>
    <param name="frame" value="tf2"/>
  </node>
  <group if="$(arg run_rviz)">
    <node name="interactive_rviz" pkg="rviz" type="rviz"
        args="-d $(find jsk_2023_09_cook_from_recipe)/rviz/view_rviz_cook.rviz" />
  </group>
</launch>
